OpenSim/Sandbox/HumphreysImplicitDynamics/forwardImplicitTest.cpp

```diff
@@ -0,0 +1,368 @@
+    Vector yDotGuess(6);
+    yDotGuess=(yPGuess-y)/h;
+
+
+    //Now let's get the residuals
+    // Muscle1's Residuals
+    const VandenBogert2011Muscle::ImplicitResidual muscle1Residual =muscle1->calcImplicitResidual(s, yDotGuess[0], yDotGuess[1], u[0],0);
+    //Muscle2's Residuals
+    const VandenBogert2011Muscle::ImplicitResidual muscle2Residual =muscle2->calcImplicitResidual(s, yDotGuess[2], yDotGuess[3], u[1],0);
+
+    // Mass Residual
+    Vec2 muscleForces = {muscle1Residual.forceTendon, muscle2Residual.forceTendon};
+    bodyResidual massResidual=calcBodyImplicitResidual(osimModel, s, yDotGuess[4], yDotGuess[5], muscleForces);
+
+    // Residual Vector
+        SimTK::Vector residuals(6);
+         residuals[0]=muscle1Residual.forceResidual,
+         residuals[1]=muscle1Residual.activResidual,
+         residuals[2]=muscle2Residual.forceResidual,
+         residuals[3]=muscle2Residual.activResidual,
+         residuals[4]=massResidual.forceResidual,
+         residuals[5]=massResidual.velocityResidual;
+
+        return residuals;
+}
+
+
+
+
+
+// =============================================================================
+// ImplicitSystemDerivativeSolver
+//        (adapted from Chris's testImplicitDifferentialEquations.cpp)
+// =============================================================================
+
+class ImplicitSystemDerivativeSolver : public OptimizerSystem {
+public:
+
+    /* Constructor class. Parameters passed are accessed in the objectiveFunc() class. */
+    ImplicitSystemDerivativeSolver(int numParameters, State& s, Model& model, Vector& u, double h):
+            OptimizerSystem(numParameters),
+            numControls(numParameters),
+            si(s),
+            osimModel(model),
+            excitation(u),
+            timestep(h)
+    {}
+
+    int objectiveFunc(const Vector &new_yDotGuess, bool new_params, Real& f) const override {
+        // No objective function
+        f = 0;
+        return 0;
+    }
+
+    int constraintFunc(const Vector &new_yDotGuess, bool newParams, Vector& constraints) const override {
+        //The constraints function is just residuals
+        Vector cons=computeImplcitResiduals(osimModel,si,new_yDotGuess,excitation,timestep);
+        constraints=cons;
+        return 0;
+    }
+
+private:
+    int numControls;
+    State& si;
+    Model& osimModel;
+    Vector& excitation;
+    double timestep;
+};
+
+
+
+
+
+// Perform a forward dynamic step
+SimTK::Vector forwardImplict(Model& osimModel, State& s, Vector yPGuess, Vector u, double h) {
+
+    int numControls =6;
+
+    // There must be a better way of doing this instead of creating a new
+    // solver at each time step (how do I handle u)
+    ImplicitSystemDerivativeSolver sys(6, s, osimModel, u, h);
+
+    Vector lower_bounds(6);
+    lower_bounds[0]=0.0;
+    lower_bounds[1]=0.0;
+    lower_bounds[2]=0.0;
+    lower_bounds[3]=0.0;
+    lower_bounds[4]=-1.0;
+    lower_bounds[5]=-SimTK::Infinity;
```

Comment:
In the eventual test, could you give a comment saying what each of these parameters are by name?